resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -446,6 +470,11 @@ private[spark] class KubernetesClusterSchedulerBackend(
                   .get()
                 val nodeName = runningExecutorPod.getSpec.getNodeName
                 val shufflePodIp = shufflePodCache.get.getShufflePodForExecutor(nodeName)
+
+                // Inform the shuffle pod about this application so it can watch.
+                kubernetesExternalShuffleClient.get
```

Comment:
Should this be `.foreach` and not `.get`? Ideally we should avoid using `.get` on Options, otherwise we're probably using Options incorrectly.